tests/yast2_gui/yast2_system_settings.pm

```diff
@@ -0,0 +1,89 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# Summary: Run yast2 system_settings module, attempt to add a non existing
+# driver in the PCI ID setup and validate the error message. Disable the sysrq
+# keys at the kernel settings and validate the change in the system configuration.
+# Enable sysrq and validate again.
+#
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use YuiRestClient;
+use YaST::Module;
+use y2lan_restart_common 'close_xterm';
+use utils 'zypper_call';
+use x11utils 'start_root_shell_in_xterm';
+
+my $system_settings;
+
+sub pre_run_hook {
+    my ($self) = @_;
+    $system_settings = $testapi::distri->get_system_settings();
+    $self->SUPER::pre_run_hook;
+}
+
+sub validate_sysrq_config {
+    my $expected_status = shift;
+    my %config_value    = (disable => 0,
+        enable => 1);
+    x11_start_program('xterm -geometry 160x45+5+5', target_match => 'xterm');
+    my $is_enabled = script_output("cat /proc/sys/kernel/sysrq");
+    $is_enabled = substr($is_enabled, 0, 1);
+    die "/proc/sys/kernel/sysrq has value $is_enabled , but sysrq should be $expected_status"
+      unless ($is_enabled eq $config_value{$expected_status});
+    $is_enabled = script_output("cat /etc/sysctl.d/70-yast.conf | grep sysrq");
```

Comment:
I haven't considered using this function here, because it works with test data and defined values. In order to use it, I would have to define the values for enabled and disabled sysrq and call the function for different test data, which makes it unnecessarily complicated. The mapping is used because I had issues passing the values 0/1 or true/false as arguments in a function. It's been quite some time so I don't remember all the things I tried. Plus, it makes the comprehension of calling the function very clear. The substr is used because there are occasions where the kernel configuration for enabled sysrq is not just 1 but 126 for example, depending on whether all sysrq functions are allowed or some are not.

Reply:
anything that is not 0 should mean that is activated, so for Perl should be enough.
This library admit regex, so for checking that an entry starts with '1' should work AFIR.
There is only one test suite, so adding those test data should be fine, you just need to call the function twice and point to the one you need, for example, just a draft of the idea:
```
validate_cfg_file(get_test_suite_data()->{disable_sysrq}->{config_files});
validate_cfg_file(get_test_suite_data()->{enable_sysrq}->{config_files});
```
So basically you call to the function for state A and state B.

Reply:
IMO to call the substring from the end and using indexes (second example 70-yast.conf) is not going to be more clear than have fixed test data, with the first example (/proc/sys/kernel/sysrq) might be if you need to get the first character (which is correct but I don't see the need because the range seems to be '0' and '1XX' so Perl booleans should suffice) 

There are several advantages to use libraries, for example you don't need to create code for handle the error.
But if you think that could be out of scope, could you please create a ticket? perhaps there is something to improve in the library to make it more appealing to be used (you are not the first person refusing this suggestion :) )

Reply:
The main reason why I prefer not to change it, is because the current code is ready, tested, working and needs no further effort. I will create a ticket.